cpp/daal/include/data_management/data/internal/numeric_table_sycl_soa.h

```diff
@@ -387,13 +316,15 @@ class DAAL_EXPORT SyclSOANumericTable : public SyclNumericTable
                 NumericTableFeature f = (*_ddict)[i];
                 if (f.typeSize != 0)
                 {
-                    DAAL_CHECK_STATUS_VAR(allocateArray(i, f));
+                    status |= allocateArray(i, f);
+                    DAAL_CHECK_STATUS_VAR(status);
                     _arraysInitialized++;
                 }
                 if (_arrays[i].empty())
                 {
                     freeDataMemoryImpl();
-                    return services::Status(services::ErrorMemoryAllocationFailed);
+                    status.add(services::ErrorMemoryAllocationFailed);
```

Comment:
Style: mix of status.add and status |=

Reply:
I don't think this is SDL problem :) Suggest not to change style here since we have mix not only in this place